src/plugins/condition/ConditionManager.js

```diff
@@ -210,13 +211,12 @@ export default class ConditionManager extends EventEmitter {
 
         if (resultObj) {
             let idAsString = this.openmct.objects.makeKeyString(resultObj.id);
-            let found = this.findConditionById(idAsString);
-            if (found) {
+            if (this.findConditionById(idAsString)) {
                 this.conditionResults[idAsString] = resultObj.data.result;
             }
         }
 
-        for (let i = 0, ii = conditionCollection.length - 1; i < ii; i++) {
+        for (let i = 0; i < conditionCollection.length - 1; i++) {
```

Comment:
FYI, the reason I keep doing ii=[x].length is because there is no reason to keep re-evaluating the length of the Array every time the for condition is checked.

Reply:
@shefalijoshi, ii is a reference to conditionCollection.length so defining and using ii is not saving evaluations. If conditionCollection has length 5, then ii gets evaluated to 5. If you add an element to the end of the conditionCollection array inside the loop, then ii becomes 6, because conditionCollection.length becomes 6.

Further, length is a property on arrays, not a function, so there really isn't any sort of computation happening.

ex.
```
let test = ['a','b']
test.length // 2
test[5] = 'c'
test.length // 6, not 3
```

Reply:
Sorry. I misspoke about storing the length in the var. I still think this is going to introduce confusion for a negligible if any gain.

Reply:
OK